WalletWasabi/Crypto/GroupElement.cs

```diff
@@ -59,5 +60,24 @@ public GroupElement(GEJ groupElement)
 		}
 
 		public static bool operator !=(GroupElement a, GroupElement b) => !(a == b);
+
+		public Scalar Sha256(GroupElement groupElement)
+		{
+			Guard.NotNull(nameof(groupElement), groupElement);
+			Guard.False($"{nameof(groupElement)}.{nameof(groupElement.IsInfinity)}", groupElement.IsInfinity);
+
+			Guard.False($"{nameof(IsInfinity)}", IsInfinity);
+
+			var concatenation = ByteHelpers.Combine(
+				Ge.x.ToBytes(),
```

Comment:
You use `Ge` but received and validated a `grouptElement`. This is related to @nothingmuch  first comment too.